### PR TITLE
Change the documentation of the 'reason' attribute

### PR DESCRIPTION
**Pull request by poncovka**
Update the documentation of the 'reason' attribute in the Requirement
class. The attribute should provide a message that explains why this
requirement is required.

Provide explanations for requirements of the Storage module.

### CODE REVIEW

File: pyanaconda/modules/common/structures/requirement.py [hunk 1]
```diff
@@ -65,9 +65,15 @@ def name(self, value: Str):
...
     def reason(self) -> Str:
-        """A reason of the requirement.
+        """An explanation of the requirement.
 
-        Usually a name of a component that requires this requirement.
+        Provides a message that explains why this requirement
+        is required.
+
+        For example:
+            Required to join the realm.
```
Comment (M4rtinK) [inline pyanaconda/modules/common/structures/requirement.py]: I would recommend: "Required to join a realm."

File: pyanaconda/modules/common/structures/requirement.py [hunk 2]
```diff
@@ -65,9 +65,15 @@ def name(self, value: Str):
...
+            Requested by the firewall kickstart command.
+            Necessary for the team device configuration.
```
Comment (M4rtinK) [inline pyanaconda/modules/common/structures/requirement.py]: I would recommend: "Necessary for network team device configuration."

File: pyanaconda/modules/common/structures/requirement.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/common/structures/requirement.py]: Fixed.

File: pyanaconda/modules/common/structures/requirement.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/modules/common/structures/requirement.py]: Fixed.

**Pull request merged**